### PR TITLE
RavenDB-9851 - Allow to resume live import from v3.0 and v2.x servers

### PR DESCRIPTION
**Pull request by grisha-kotler**
import attachments from v2.x servers.

http://issues.hibernatingrhinos.com/issue/RavenDB-9851

### CODE REVIEW

File: src/Raven.Server/Smuggler/Migration/MigrationConfiguration.cs [hunk 1]
```diff
@@ -21,12 +21,13 @@ public abstract class MigrationConfigurationBase
...
         public string ServerUrl { get; set; }
 
         public string UserName { get; set; }
 
         public string Password { get; set; }
 
-        public string Domain { get; set; }
-    }
+        public string Domain { get; set; }
    }
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Migration/MigrationConfiguration.cs:32]: CRLF issue

File: src/Raven.Server/Smuggler/Migration/AbstractMigrator.cs [hunk 2]
```diff
@@ -1,39 +1,5 @@
...

        public abstract void Dispose();

        protected async Task SaveLastOperationState(BlittableJsonReaderObject blittable)
        {
            var cmd = new MergedPutCommand(blittable, MigrationStateKey, null, Database);
            await Database.TxMerger.Enqueue(cmd);
        }
    }
}
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Migration/AbstractMigrator.cs:5]: CRLF issue

File: src/Raven.Server/Smuggler/Migration/AbstractMigrator.cs [hunk 3]
```diff
@@ -1,39 +1,5 @@
...
-using Raven.Server.ServerWide;
+﻿using System;
using System.Net.Http;
using System.Threading.Tasks;
using Raven.Client.Documents.Operations;
using Raven.Client.Documents.Smuggler;
using Raven.Server.Documents;
using Raven.Server.Documents.Handlers;
using Raven.Server.ServerWide;
using Sparrow.Json;
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Migration/AbstractMigrator.cs:1]: CRLF issue

File: src/Raven.Server/Smuggler/Documents/StreamSource.cs [hunk 4]
```diff
@@ -731,20 +760,7 @@ internal unsafe LegacyAttachmentDetails ProcessLegacyAttachment(DocumentsOperati
...
            public string Key;
            public long Size;
            public string Tag;
            public BlittableJsonReaderObject Metadata;
        }

        private const char RecordSeperator = (char)SpecialChars.RecordSeparator;
        private const string DummyDocumentPrefix = "files/";

        public unsafe void ProcessAttachmentStream(DocumentsOperationContext context, BlittableJsonReaderObject data, ref DocumentItem.AttachmentStream attachment)
```
Comment (ppekrol) [inline src/Raven.Server/Smuggler/Documents/StreamSource.cs:763]: CRLF issue

File: src/Raven.Client/Documents/Smuggler/SmugglerResult.cs [hunk 5]
```diff
@@ -76,7 +76,7 @@ public override DynamicJsonValue ToJson()
 
         public bool ShouldPersist => true;
 
-        public class SmugglerProgress : SmugglerProgressBase, IOperationProgress
+        public string LegacyLastDocumentEtag { get; set; }        public string LegacyLastAttachmentEtag { get; set; }        public class SmugglerProgress : SmugglerProgressBase, IOperationProgress
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Smuggler/SmugglerResult.cs:79]: CRLF issue

**Comment by ppekrol** [comment]
Hi @grisha-kotler
I see that you are pushing a lot of CR/LF stuff. I had similar problem, but updating to latest R# EAP solved it https://www.jetbrains.com/resharper/eap/

**Comment by ayende** [comment]
Manually merged

**Pull request closed without merging**